app/models/configuration_script.rb

```diff
@@ -1,3 +1,17 @@
 class ConfigurationScript < ActiveRecord::Base
-  belongs_to :configuration_manager, :class_name => 'ManageIQ::Providers::ConfigurationManager'
+  belongs_to  :manager,
+              :class_name  => 'ManageIQ::Providers::ConfigurationManager',
+              :foreign_key => :configuration_manager_id
+  include ProviderObjectMixin
+
+  def run(vars=nil)
+    extra_vars = vars.reverse_merge(variables)
```

Comment:
@syncrou 
The Job Template has an option to store extra vars either as a YAML or JSON, so if the customer were to create the Job Template with the extra vars in YAML and use it with ManageIQ, we should be able to convert the YAML to JSON and then forward it to the Tower Client.

Also should ConfigurationScript#run support simple Hashes along with YAML and JSON strings.